server/middleware/middleware.go

```diff
@@ -78,21 +78,20 @@ func Get(config *config.Config, tenantMgr *metadata.TenantManager, txMgr *transa
 		Logger()
 
 	// The order of the interceptors matter with optional elements in them
-	streamInterceptors := []grpc.StreamServerInterceptor{
-		forwarderStreamServerInterceptor(),
-	}
+	var streamInterceptors []grpc.StreamServerInterceptor
 
-	if config.Auth.Enabled {
-		streamInterceptors = append(streamInterceptors, grpc_auth.StreamServerInterceptor(authFunction))
+	if config.Tracing.Enabled {
+		streamInterceptors = append(streamInterceptors, traceStream())
 	}
 
-	streamInterceptors = append(streamInterceptors, namespaceInitializer.NamespaceSetterStreamServerInterceptor())
+	streamInterceptors = append(streamInterceptors, forwarderStreamServerInterceptor())
 
-	if config.Tracing.Enabled {
-		streamInterceptors = append(streamInterceptors, traceStream())
+	if config.Auth.Enabled {
```

Comment:
Should we make tracing granularly configurable? Such as, make it feasible to turn ot on/off for other components in case there's measurable or significant overhead?

Reply:
We eventually should, in a separate pull request. This is just refactoring and making auth part of it.

Reply:
But even if it was more granuarly configurable, loading the top level interceptor would be needed if there is any kind of tracing enabled.